### PR TITLE
remove npm as a dependency

### PR DESCRIPTION
**Pull request by charlierudolph**
@alexdavid @kevgo

Use the global npm for the install to avoid version differences (2 vs 3)

### CODE REVIEW

File: src/mycha_installer.coffee [hunk 1]
```diff
@@ -19,10 +19,13 @@ class MychaInstaller
...
 
   _installDependencies: (done) =>
-    @_writeToStdout colors.bold('npm install --save-dev chai sinon sinon-chai'), lineBefore: yes
-    npm.load 'save-dev': yes, (err) =>
-      if err then return done err
-      npm.commands.install @cwd, ['chai', 'sinon', 'sinon-chai'], done
+    cmd = 'npm'
+    args = ['install', '--save-dev', 'chai', 'sinon', 'sinon-chai']
+    cmdString = [cmd].concat(args).join ' '
+    @_writeToStdout colors.bold(cmdString), lineBefore: yes
```
Comment (kevgo) [inline src/mycha_installer.coffee]: Maybe get rid of the `cmdString` temp variable here? It doesn't add readability in my opinion.

```coffeescript
@_writeToStdout colors.bold([cmd].concat(args).join ' '), lineBefore: yes

**Comment by kevgo** [comment]
Much cleaner and nicer! LGTM from my end

File: src/mycha_installer.coffee (hunk 1 above)
Comment (alexdavid) [inline src/mycha_installer.coffee]:
```coffeescript
@_writeToStdout colors.bold([cmd, args...].join ' '), lineBefore: yes
```

File: features/support/world.coffee [hunk 2]
```diff
@@ -5,6 +6,13 @@ class World
 
   mychaPath: path.join __dirname, '..', '..', 'bin', 'mycha'
 
+
+  installMycha: (testHelperPath, done) ->
+    childProcess = fork @mychaPath, ['install'], silent: yes, cwd: @tmpDir
+    childProcess.stdin.write "#{testHelperPath}\n", 'utf-8'
```
Comment (alexdavid) [inline features/support/world.coffee]: This is just blindly filling in the test helper path prompt, right?

What do you think about making `mycha install` accept a command line argument `--testHelperPath` and only prompt the user if it's not provided?

Comment (alexdavid) [inline features/support/world.coffee]: On second thought, this is probably fine how it is

**Pull request closed without merging**